src/components/loginform/loginform.js

```diff
@@ -1,106 +1,149 @@
+          });
+        } else {
+          this.setState({
+            passwordError: null,
+          });
+        }
+        break;
+      }
     }
-    return false;
+    return;
   };
-  onSubmit = (e) => {
+
+  onFormSubmit = (e) => {
     e.preventDefault();
-    let requestBody = {
-      query: `
-        query {
-          login(email: "${this.state.email}", password: "${this.state.password}") {
-            userId
-            token
-            tokenexpiration
-            username
-          }
-        }
-      `,
-    };
-    fetch("http://localhost:8000/graphql", {
-      method: "POST",
-      body: JSON.stringify(requestBody),
-      headers: { "Content-Type": "application/json" },
-    })
-      .then((res) => {
-        if (res.status !== 200 && res.status !== 201) {
-          throw new Error("Failed");
-        }
-        return res.json();
-      })
-      .then((resData) => {
-        Login(
-          resData.data.login.token,
-          resData.data.login.tokenexpiration,
-          resData.data.login.userId,
-          resData.data.login.username
-        );
-        console.log(resData)
-        this.props.handleLogin();
-      })
-      .catch((err) => {
-        throw new Error(err);
-      });
+    this.props.login(this.state.email, this.state.password);
   };
 
+  static getDerivedStateFromProps(props, state) {
+    if (props.error) {
+      return {
+        formSubmissionError: props.error,
+      };
+    }
+    return null;
+  }
+
+  componentDidUpdate() {
+    if (this.state.isFormInvalid != false) {
+      if (this.state.emailError === null && this.state.passwordError === null) {
+        this.setState({ isFormInvalid: false });
+      }
+    } else {
+      if (this.state.emailError !== null || this.state.passwordError !== null)
+        this.setState({ isFormInvalid: true });
+    }
+  }
+
   render() {
     return (
-      <div className="login-details">
-        <Form onSubmit={this.onSubmit}>
+      <div className="login-form">
+        {this.state.formSubmissionError && (
+          <div className="alert alert-danger" role="alert">
+            {this.state.formSubmissionError}
+          </div>
+        )}
+        <Form onSubmit={this.onFormSubmit}>
           <Form.Group controlId="formBasicEmail">
             <Form.Label>
               <div className="formLabel">Email address</div>
             </Form.Label>
             <Form.Control
-              onChange={(e) => this.handleChange(e, "email")}
+              onChange={this.onFieldChange}
               type="email"
-              placeholder="abc@gmail.com"
+              name="email"
             />
+            {this.state.emailError && (
+              <h6 className="form-field-error">{this.state.emailError}</h6>
+            )}
           </Form.Group>
           <Form.Group controlId="formBasicPassword">
             <Form.Label>
               <div className="formLabel">Password</div>
```

Comment:
`className="formLabel"`  Maybe use a single naming convention?
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

Reply:
Fixed. Going with hyphenated ones